nvme-print-json.c

```diff
@@ -47,17 +47,51 @@
 static const uint8_t zero_uuid[16] = { 0 };
 static struct print_ops json_print_ops;
 
-static const char *supported = "Supported";
-static const char *not_supported = "Not Supported";
-static const char *yes = "Yes";
-static const char *no = "No";
-static const char *enabled = "Enabled";
-static const char *disabled = "Disabled";
-static const char *reserved = "Reserved";
-static const char *true_str = "True";
+static const char *address_str = "Address";
+static const char *controller_str = "Controller";
+static const char *count_str = "Count";
+static const char *data_str = "Data";
+static const char *device_str = "Device";
+static const char *disabled_str = "Disabled";
+static const char *element_str = "Element";
+static const char *enabled_str = "Enabled";
+static const char *endry_str = "Entry";
+static const char *error_str = "Error";
 static const char *false_str = "False";
-static const char *result_str = "result";
-static const char *error_str = "error";
+static const char *feature_str = "Feature";
+static const char *firmware_str = "Firmware";
+static const char *generic_str = "Generic";
+static const char *host_id_str = "Host ID";
+static const char *length_str = "Length";
+static const char *model_number_str = "Model number";
+static const char *name_str = "Name";
+static const char *namespace_str = "Namespace";
+static const char *no_str = "No";
+static const char *not_enabled_str = "Not enabled";
+static const char *not_supported_str = "Not supported";
+static const char *not_valid_str = "Not valid";
+static const char *nsid_str = "NSID";
+static const char *operation_str = "Operation";
+static const char *path_str = "Path";
+static const char *port_id_str = "Port ID";
+static const char *reserved_str = "Reserved";
+static const char *result_str = "Result";
+static const char *revision_str = "Revision";
+static const char *state_str = "State";
+static const char *serial_number_str = "Serial number";
+static const char *slot_str = "Slot";
+static const char *status_str = "Status";
+static const char *subsystem_str = "Subsystem";
+static const char *supported_str = "Supported";
+static const char *temperature_str = "Temperature";
+static const char *threshold_str = "Threshold";
+static const char *transport_str = "Transport";
+static const char *true_str = "True";
+static const char *type_str = "Type";
+static const char *unknow_str = "Unknown";
+static const char *value_str = "Value";
+static const char *version_str = "Version";
+static const char *yes_str = "Yes";
```

Comment:
I am not really convinced we should start to move these strings out. These days, the compiler and linker are very smart and are able to avoid data duplication. And these are only a few of the keys and we end up with things like:

```c
					obj_add_str(jns, namespace_str, nvme_ns_get_name(n));
					obj_add_str(jns, generic_str, nvme_ns_get_generic_name(n));
					obj_add_int(jns, nsid_str, nvme_ns_get_nsid(n));
					obj_add_uint64(jns, "UsedBytes", nuse);
					obj_add_uint64(jns, "MaximumLBA", nvme_ns_get_lba_count(n));
					obj_add_uint64(jns, "PhysicalSize", nsze);
```

two styles as of key. Which I really do not like because it's not clear when to use what. I assume if a key used more than once use an `const char *` definition.

Sure the argument that allows to update the string in one place is nice but I don't think we have a huge problem here.

In other words I think we should not do it at all. Could you drop this patch and also remove the already existing `const char *` variables? Thanks!

Reply:
Thanks for your comments. Just fixed by the additional commit [7a7b142](https://github.com/linux-nvme/nvme-cli/pull/2088/commits/7a7b142d7704f4bf430160bb7de0c7d375b9779c) but not deleted the original commit since seems some other commits are depended on the original commit. If needed will do delete the original commit.